cleanup inconsistencies and other minor issues in the readme

zipsegv opened the pull request.

zipsegv committed:
cleanup readme

NSGolova commented on README.md:
```diff
@@ -4,11 +4,7 @@ Beat Saber open replay format.
 # BSOR V1
 
-- [C# code](https://github.com/BeatLeader/beatleader-mod/blob/master/Source/2_Core/Models/Replay.cs)
-- [JS code](https://github.com/BeatLeader/BeatSaber-Web-Replays/blob/master/src/open-replay-decoder.js)
```
Why remove all the non-C# implementations?

TheZipCreator replied:
the links to them don't work anymore and I couldn't find the code in the repo

NSGolova commented on README.md:
```diff
@@ -17,31 +13,31 @@ Beat Saber open replay format.
-  trackingSytem;               - string, tracking system type. (OpenVR, Oculus, etc.)
-  hmd;                         - string, headset type. (Oculus Quest, Valve Index, etc.)
-  controller;                  - string, controllers type. (Oculus touch, etc)
+  trackingSytem;               - string, tracking system type. (OpenVR, Oculus, etc.).
```
2 dots?

NSGolova commented on README.md:
```diff
@@ -50,27 +46,27 @@ Beat Saber open replay format.
+  {                            - Head structure.
     {x, y, z}                  - 3 floats, position.
-    {x, y, z, w}               - 4 floats, rotation.
+    {x, y, z, w}               - 4 floats, rotation (in quaternion form).
```
idk if it's necessary to specify when rotation has 4 floats and there are few more quaternions further the file

NSGolova commented on README.md:
```diff
@@ -80,63 +76,63 @@ noteCount                      - int, note events count.
+    { x, y, z }                - 3 floats, cut normal vector.
+    cutDistanceToCenter        - float, distance from center of the cut.
+    cutAngle                   - float, angle of the cut.
+    beforeCutRating            - float.
```
Angle rating before the cut, with after cut and cut distance forming 115 score. This beforeCutRating is uncapped (can go over 1) 1 means 70 score.

NSGolova commented on README.md:
```diff
@@ -80,63 +76,63 @@ noteCount                      - int, note events count.
+    cutDistanceToCenter        - float, distance from center of the cut.
+    cutAngle                   - float, angle of the cut.
+    beforeCutRating            - float.
+    afterCutRating             - float.
```
Angle rating after the cut, with before cut and cut distance forming 115 score. This afterCutRating is uncapped (can go over 1) 1 means 30 score.

NSGolova commented on README.md:
```diff
@@ -80,63 +76,63 @@ noteCount                      - int, note events count.
+    cutDirDeviation            - float, how far away the cut direction was from the one expected.
+    { x, y, z }                - 3 floats, point at which the note was cut.
+    { x, y, z }                - 3 floats, cut normal vector.
+    cutDistanceToCenter        - float, distance from center of the cut.
```
distance from the center of the note to the cut plane, to get the score `15 * (1 - Clamp01(cutDistanceToCenter / 0.3f))`. With before and after cuts forming 115 score

NSGolova commented on README.md:
```diff
@@ -80,63 +76,63 @@ noteCount                      - int, note events count.
+    saberTypeOK                - bool, note was hit with correct saber.
+    wasCutTooSoon              - bool, note was cut too soon.
+    saberSpeed                 - float, speed of the saber.
+    { x, y, z }                - 3 floats, direction of the saber.
```
Let's preserve the name of the property, as it's more specific and corresponds to the game name
`{ x, y, z }                - 3 floats, saberDir, direction of the saber.`

NSGolova commented on README.md:
```diff
@@ -80,63 +76,63 @@ noteCount                      - int, note events count.
+    saberType                  - int, type of the saber (0 = left, 1 = right).
+    timeDeviation              - float, how far away the time hit was from the time expected.
+    cutDirDeviation            - float, how far away the cut direction was from the one expected.
+    { x, y, z }                - 3 floats, point at which the note was cut.
```
same with these 2

zipsegv committed:
further fix readme

zipsegv commented:
regarding removing the non-C# implementations, the C++ link was broken and I assumed the other ones were too without actually checking. I've now reinstated the other implementations and fixed the C++ link.
I also changed where rotations being quaternions are mentioned, and then I've also followed your suggestions.
